### PR TITLE
Send Mass Welcome Email to Users

### PR DESCRIPTION
**Pull request by ascii-dev**
## Description
Sends mass welcome email with reset password link to users.

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: gmmp/templates/emails/welcome_password_reset.html [hunk 1]
```diff
@@ -0,0 +1,422 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta name="viewport" content="width=device-width" />
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+    <title>Simple Transactional Email</title>
```
Comment (esirK) [inline gmmp/templates/emails/welcome_password_reset.html]: we can change this title.

Comment (ascii-dev) [reply gmmp/templates/emails/welcome_password_reset.html]: Yes... I am not fully through with the template yet as it's not being rendered well in the email. As soon as that's done, I'll fix the write up. @esirK

File: gmmp/management/commands/bulk_password_reset_emails.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        users = User.objects.all()
+        datatuple = []
+        password_reset_form = render_to_string(
+            'emails/welcome_password_reset.html', )
+        for user in users:
+            datatuple.append(
+                (
+                    'Reset your password on GMMP',
+                    password_reset_form,
+                    'from@gmmp.com',
```
Comment (esirK) [inline gmmp/management/commands/bulk_password_reset_emails.py]: not sure if this should be `from@gmmp.com`

Comment (ascii-dev) [reply gmmp/management/commands/bulk_password_reset_emails.py]: I know! Still working on rendering the email correctly first as that takes more priority...

**Review by kilemensi — COMMENTED** [review]
:+1: @ascii-dev 

This is how password reset email looks like at the moment:
Screenshot from 2020-09-28 16-41-48

1. It's fully text (no html),
2. And uses default envs set e.g. from is webmaster@localhost, etc.

Can't we do something very similar to this i.e. no html/styling issues? The only things that need to be different is the:

1. First paragraph: `You've received this email ...` should change to something like `Welcome to GMMP 2020. Your new account has been successful created at app.gmmp.ngo`. The second paragraph and link can remain the same i.e. `Please go to ..`
2. Your username paragraph: The `Your username, in case ...` need a small tweak  by removing the `, in case you've forgotten` since it's a brand you account.

Then done! Simple & effect.

**Comment by ascii-dev** [comment]
> 👍 @ascii-dev
> 
> This is how password reset email looks like at the moment:
> Screenshot from 2020-09-28 16-41-48
> 
> 1. It's fully text (no html),
> 2. And uses default envs set e.g. from is webmaster@localhost, etc.
> 
> Can't we do something very similar to this i.e. no html/styling issues? The only things that need to be different is the:
> 
> 1. First paragraph: `You've received this email ...` should change to something like `Welcome to GMMP 2020. Your new account has been successful created at app.gmmp.ngo`. The second paragraph and link can remain the same i.e. `Please go to ..`
> 2. Your username paragraph: The `Your username, in case ...` need a small tweak  by removing the `, in case you've forgotten` since it's a brand you account.
> 
> Then done! Simple & effect.

I think this one's for when a user has gone to the application to start the password reset process. What I've been working on is the email that we'll send to prompt them to go start the password request process. @kilemensi

**Comment by kilemensi** [comment]
> I think this one's for when a user has gone to the application to start the password reset process. What I've been working on is the email that we'll send to prompt them to go start the password request process. @kilemensi

I know that @ascii-dev ... I'm saying lets make it _similar_. If the reset one is just text and all the users understand it, why do we need to change now?

... _if it ain't broke, don't fix it._

**Comment by ascii-dev** [comment]
> > I think this one's for when a user has gone to the application to start the password reset process. What I've been working on is the email that we'll send to prompt them to go start the password request process. @kilemensi
> 
> I know that @ascii-dev ... I'm saying lets make it _similar_. If the reset one is just text and all the users understand it, why do we need to change now?
> 
> ... _if it ain't broke, don't fix it._

@kilemensi Oh... I get what you're saying now... The HTML issues has even been fixed. The problem I'm just facing now is referencing the `absolute url path` in the password reset link without the `request context`. I'm thinking of just moving the code from the `management command` into a view so we can have access to the `request context`.

File: .gitignore [hunk 3]
```diff
@@ -117,7 +117,7 @@ celerybeat.pid
 *.sage.py
 
 # Environments
-.env
+.env/.env
```
Comment (kilemensi) [inline .gitignore]: What's the reason for this change?

File: gmmp/management/commands/bulk_password_reset_emails.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        emails.append(email)
+    return get_connection().send_messages(emails)
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        users = User.objects.all()
+        password_reset_form = render_to_string(
+            'emails/welcome_password_reset.html', {'SITE_URL': settings.SITE_URL})
+        send_mass_html_mail(
+            'Reset your password on GMMP',
```
Comment (kilemensi) [inline gmmp/management/commands/bulk_password_reset_emails.py]: This should probably be `New account on app.gmmp.ngo` ... Otherwise it may send a confusing message to those with accounts already.

File: gmmp/management/commands/bulk_password_reset_emails.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            '''Hi there''',
```
Comment (kilemensi) [inline gmmp/management/commands/bulk_password_reset_emails.py:24]: Where does this go?

File: gmmp/settings.py [hunk 6]
```diff
@@ -158,10 +158,15 @@
...
 
+
+SITE_URL = env.get('SITE_URL', 'http://localhost:8000')
+
+
 # Temporary login screen for downloading the exported data
 LOGIN_URL = "/admin/login/"
 
 # Email
+FROM_EMAIL = env.get("FROM_EMAIL", "support@gmmp.ngo")
```
Comment (kilemensi) [inline gmmp/settings.py]:
```suggestion
EMAIL_FROM = env.get("GMMP_EMAIL_FROM", "webmaster@localhost")
```

To be consistent with other email variables/site_url defaulting to localhost

File: gmmp/templates/emails/welcome_password_reset.html [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+                    cellpadding="0"
+                    cellspacing="0"
+                  >
+                    <tr>
+                      <td>
+                        <p>Hi there,</p>
+                        <p>
+                            A GMMP account has been created for you. To access this account,
+                            kindly reset your password by clicking on the button below:
+                        </p>
```
Comment (kilemensi) [inline gmmp/templates/emails/welcome_password_reset.html]:
```suggestion
                        <h3>Hello,</h3>
                        <p>
                            A new account has been set up for you on app.gmmp.ngo. To access this account,
                            kindly reset your password by clicking on the button below:
                        </p>
```

File: gmmp/templates/emails/welcome_password_reset.html [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+                                        >
+                                      </td>
+                                    </tr>
+                                  </tbody>
+                                </table>
+                              </td>
+                            </tr>
+                          </tbody>
+                        </table>
+                        <p>Good luck!</p>
```
Comment (kilemensi) [inline gmmp/templates/emails/welcome_password_reset.html]:
```suggestion
```
Why do they need luck?

Also shouldn't we include username information with this email?

**Review by kilemensi — COMMENTED** [review]
:+1: @ascii-dev 

-----------------

Cool, so I we combining this with #97 or not?

cc @esirK

**Comment by esirK** [comment]
> 👍 @ascii-dev
> 
> Cool, so I we combining this with #97 or not?
> 
> cc @esirK

I think let's keep them as 2 different commands.

File: .gitignore (hunk 3 above)
Comment (esirK) [reply .gitignore]: We should leave it as `.env` to prevent exposing the `credentials.json` here https://github.com/CodeForAfrica/gmmp/blob/master/gmmp/settings.py#L299

Comment (ascii-dev) [reply .gitignore]: I made the change so that we can include the `.env.template` which will act as a guide for what the data that the `.env` file needs. @kilemensi cc @esirK

File: gmmp/management/commands/bulk_password_reset_emails.py (hunk 5 above)
Comment (ascii-dev) [reply gmmp/management/commands/bulk_password_reset_emails.py:24]: Don't know why it's required. My thoughts are it's for a email clients that can't process `html` but I might  be wrong. @kilemensi

File: gmmp/templates/emails/welcome_password_reset.html (hunk 8 above)
Comment (ascii-dev) [reply gmmp/templates/emails/welcome_password_reset.html]: > Why do they need luck?

It was just a little fun idea that snuck into my head while editing the template. 😼 

> Also shouldn't we include username information with this email?

Thought about it but decided to keep `KISS`

@kilemensi

**Review by kilemensi — APPROVED** [review]
